File: src/frontend/src/types/complaint.ts
export type IssueType = 'garbage_accumulation' | 'overflowing_bin' | 'construction_debris' | 'pothole';

export interface Complaint {
  id: number;
  complaint_number: string;
  issue_type: IssueType;
  description: string;
  latitude: number;
  longitude: number;
  address: string;
  status: string;
  created_at: string;
  assigned_worker_id?: number;
  service_area?: string;
  worker_acknowledged_at?: string;
}

export interface Evidence {
  id: number;
  complaint_id: number;
  type: 'BEFORE' | 'AFTER' | 'WORK_PROOF' | 'FIELD_INSPECTION';
  file_path: string;
  timestamp: string;
  latitude?: number;
  longitude?: number;
  quality_score?: number;
}

export interface ComplaintCreateInput {
  issue_type: IssueType;
  description: string;
  latitude: number;
  longitude: number;
  address: string;
}

File: src/backend/app/schemas/complaint.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ComplaintBase(BaseModel):
    issue_type: str
    description: Optional[str] = None
    latitude: float
    longitude: float
    address: Optional[str] = None
    authority: Optional[str] = None
    created_by: Optional[int] = None

class ComplaintCreate(ComplaintBase):
    pass

class ComplaintResponse(ComplaintBase):
    id: int
    complaint_number: str
    status: str
    assigned_worker_id: Optional[int] = None
    service_area: Optional[str] = None
    worker_acknowledged_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True
